src/core/dfn.js

```diff
@@ -9,13 +9,9 @@ export const name = "core/dfn";
 
 export function run() {
   document.querySelectorAll("dfn").forEach(dfn => {
-    /** @type {HTMLElement} */
-    const closestDfn = dfn.closest("[data-dfn-for]");
-    if (closestDfn && closestDfn !== dfn && !dfn.dataset.dfnFor) {
-      dfn.dataset.dfnFor = closestDfn.dataset.dfnFor;
```

Comment:
I feel this might affect Shepherd to miss `data-dfn-for`, are we sure this is okay?

Reply:
Ah, good point. We might indeed need to transplant data-dfn-for explicitly.

Reply:
Note, we can't do this for all definitions yet... only for IDL ones for now, because concept definitions are currently not "for" anything.

Reply:
> Note, we can't do this for all definitions yet... only for IDL ones for now, because concept definitions are currently not "for" anything.

IDL ones get their `data-dfn-for` from `decorateDfn()` which means this should be okay. 👍

Reply:
ok cool. Remove this conversation?

Reply:
Oh wait, why do we need ``dfn.closest(`[data-dfn-for="${parent}"]`)`` then?

Reply:
having a look...

Reply:
Do you mean in src/core/dfn-finder.js? that's exclusively used by core/webidl.

Reply:
Ah yes, `findDfn()` uses `closest()` to find definitions and then assign the right value by `decorateDfn()`. 👍